google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/DdlBatch.java

```diff
@@ -0,0 +1,211 @@
+  }
+
+  @Override
+  public Timestamp getReadTimestampOrNull() {
+    return null;
+  }
+
+  @Override
+  public Timestamp getCommitTimestamp() {
+    throw SpannerExceptionFactory.newSpannerException(ErrorCode.FAILED_PRECONDITION,
+        "There is no commit timestamp available for DDL batches.");
+  }
+
+  @Override
+  public Timestamp getCommitTimestampOrNull() {
+    return null;
+  }
+
+  @Override
+  public void executeDdl(ParsedStatement ddl) {
+    ConnectionPreconditions.checkState(state == UnitOfWorkState.STARTED,
+        "The batch is no longer active and cannot be used for further statements");
+    Preconditions.checkArgument(ddl.getType() == StatementType.DDL,
+        "Only DDL statements are allowed. \"" + ddl.getSqlWithoutComments()
+            + "\" is not a DDL-statement.");
+    statements.add(ddl.getSqlWithoutComments());
+  }
+
+  @Override
+  public long executeUpdate(ParsedStatement update) {
+    throw SpannerExceptionFactory.newSpannerException(ErrorCode.FAILED_PRECONDITION,
+        "Executing updates is not allowed for DDL batches.");
+  }
+
+  @Override
+  public long[] executeBatchUpdate(Iterable<ParsedStatement> updates) {
+    throw SpannerExceptionFactory.newSpannerException(ErrorCode.FAILED_PRECONDITION,
+        "Executing batch updates is not allowed for DDL batches.");
+  }
+
+  @Override
+  public void write(Mutation mutation) {
+    throw SpannerExceptionFactory.newSpannerException(ErrorCode.FAILED_PRECONDITION,
+        "Writing mutations is not allowed for DDL batches.");
+  }
+
+  @Override
+  public void write(Iterable<Mutation> mutations) {
+    throw SpannerExceptionFactory.newSpannerException(ErrorCode.FAILED_PRECONDITION,
+        "Writing mutations is not allowed for DDL batches.");
+  }
+
+  /**
+   * Create a {@link ParsedStatement} that we can use as input for the generic execute method
+   * when the {@link #runBatch()} method is executed. This method uses the generic execute
+   * method that allows statements to be cancelled and to timeout, which requires the input to be
+   * a {@link ParsedStatement}.
+   */
+  private static final ParsedStatement RUN_BATCH =
+      StatementParser.INSTANCE.parse(Statement.of("RUN BATCH"));
+
+  @Override
+  public long[] runBatch() {
+    ConnectionPreconditions.checkState(state == UnitOfWorkState.STARTED,
+        "The batch is no longer active and cannot be ran");
+    try {
+      if (!statements.isEmpty()) {
+        // create a statement that can be passed in to the execute method
+        Callable<Void> callable = new Callable<Void>() {
+          @Override
+          public Void call() throws Exception {
+            OperationFuture<Void, UpdateDatabaseDdlMetadata> operation =
+                ddlClient.executeDdl(statements);
+            return operation.get();
+          }
+        };
+        asyncExecuteStatement(RUN_BATCH, callable);
+      }
+      this.state = UnitOfWorkState.RAN;
+      return new long[0];
```

Comment:
Shouldn't this return the number of successful updates?

Reply:
Changed to return an array containing an indication per statement whether the statement was successfully executed (1) or not (0).